hack/v1migration/phase1.sh

```diff
@@ -0,0 +1,50 @@
+#!/bin/sh
+
+set -e
+
+if [ $# != 1 ]; then
+	echo "USAGE: $0 [WORKDIR]"
+	exit 1
+fi
+
+WORKDIR=$1
+echo "Using workdir: $WORKDIR"
+
+oc project hive
+
+oc scale -n hive deployment.v1.apps/hive-operator --replicas=0
+oc scale -n hive deployment.v1.apps/hive-controllers --replicas=0
+oc scale -n hive deployment.v1.apps/hiveadmission --replicas=0
+
+# TODO: wait until no pods are running
+
+mkdir -p $WORKDIR/
+
+HIVE_TYPES=( checkpoints clusterdeployments clusterdeprovisionrequests clusterimagesets clusterprovisions clusterstates dnsendpoints dnszones hiveconfig selectorsyncidentityprovider selectorsyncset syncidentityprovider syncsetinstance syncset )
+for i in "${HIVE_TYPES[@]}"
+do
+	:
+	echo "Storing all ${i} in ${WORKDIR}/${i}.yaml"
+	oc get ${i}.hive.openshift.io -A -o json | jq '.items | .[] |
+		del(.status) |
+		del(.metadata.annotations."kubectl.kubernetes.io/last-applied-configuration") |
+		del(.metadata.creationTimestamp) |
+		del(.metadata.generation) |
+		del(.metadata.resourceVersion) |
+		del(.metadata.ownerReferences) |
+		del(.metadata.selfLink) |
+		del(.metadata.uid)' > ${WORKDIR}/${i}.yaml
+done
+
```

Comment:
I would like an added check here that fetches the resources again and compares them to what is stored in the yaml before proceeding.

Reply:
What is the thought behind this one?

Reply:
I may have figured out why on the other script but is there still a reason to do this here, immediately after fetching the yaml, but before we've even removed finalizers?

Reply:
It would be a safety check that verifies that everything was correctly stored in the files before we start doing destructive things.